javaparser-core/src/main/java/com/github/javaparser/parser/ASTParserBase.java

```diff
@@ -0,0 +1,112 @@
+package com.github.javaparser.parser;
+
+import com.github.javaparser.Position;
+import com.github.javaparser.ast.body.ModifierSet;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author Didier Villevalois
+ */
+public abstract class ASTParserBase {
+
+
+	protected List add(List list, Object obj) {
+		if (list == null) {
+			list = new LinkedList();
```

Comment:
Yep. I did not modify too much the original code, just extracted it yet. In fact, I plan to do more work on that stuff to simplify the grammar further and minimize the ammount of java code in the grammar.